src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -703,4 +830,211 @@ export class NetworkAreaDiagramViewer {
             converterStationElement.querySelector('polyline');
         polyline?.setAttribute('points', polylinePoints);
     }
+
+    private moveLoopEdgeGroup(
+        edges: SVGGraphicsElement[],
+        mousePosition: Point
+    ) {
+        edges.forEach((edge) => {
+            // get edge element
+            const edgeId = edge.getAttribute('svgid');
+            if (!edgeId) {
+                return;
+            }
+            const translation = new Point(
+                mousePosition.x - this.initialPosition.x,
+                mousePosition.y - this.initialPosition.y
+            );
+            this.moveSvgElement(edgeId, translation);
+        });
+    }
+
+    private storeHalfEdgeAngle(
+        edgeNode: SVGGraphicsElement,
+        angleId: string,
+        sideIndex: number,
+        getAngle: (a: HTMLElement) => number | null
+    ) {
+        if (!this.edgeAngles.has(angleId)) {
+            const halfEdge: HTMLElement = <HTMLElement>(
+                edgeNode.children.item(sideIndex)?.firstElementChild
+            );
+            if (halfEdge != null) {
+                const angle = getAngle(halfEdge);
+                if (angle != null) {
+                    this.edgeAngles.set(angleId, angle);
+                }
+            }
+        }
+    }
```

Comment:
this is now only called in `getEdgeAngle`, so we could remove this method and include it there directly.

Besides we can remove the `if (!this.edgeAngles.has(angleId))`, as it is already checked before in `getEdgeAngle`

Reply:
Code moved to getEdgeAngle method